### DISCUSSION TITLE
Does Parcel support css isolation?

### DISCUSSION THREAD

**Discussion by Sossenbinder** [parcel-bundler/parcel#6151] [Q&A]
Hello,

I'm currently developing a React application with Parcel and have a great time with it honestly.

However, I'm wondering whether CSS isolation is available, and if yes, how?

My workflow is usually to create one .less file for each React component I'm writing, and then import it within the component. So for example:

```
import * as React from "react";

import "./Styles/LoadingBar.less";

type Props = {

}

export const LoadingBar: React.FC<Props> = () => {
	return (
		<div className={"Container"}>
			...
		</div>
	);
}

export default LoadingBar;
```

Now, this works, however, if the "Container" class is defined in any other .less file for another component as well, then these two are not isolated, and the styles will overlap.

Is it possible to somehow configure the classes to only match to the respective component? I've seen this done with compile-time transforms of class names, appending a unique ID to the respective class scope.

**Comment by mischnic** [comment]
You could use CSS modules:

https://v2.parceljs.org/languages/postcss/#css-modules

**Reply by Sossenbinder** [reply]
That is postcss only though, correct?

**Reply by mischnic** [reply]
Parcel will handle this for you, LESS will first be transpiled to CSS, which is then processed with PostCSS as necessary. So
```js
import { main } from "./app.module.less";
```
should work fine
